src/ostorlab/agent/mixins/agent_report_vulnerability_mixin.py

```diff
@@ -3,11 +3,55 @@
 Definition of the main method to process an entry from the KB - Knowledge Base,
 and emit a vulnerability message.
 """
-from typing import Optional
+import dataclasses
 import enum
+from typing import Optional, List
 
-from ostorlab.agent.mixins.protocols import emit
 from ostorlab.agent.kb import kb
+from ostorlab.agent.mixins.protocols import emit
+from ostorlab.assets import asset
+
+MAP_ASSETS_NAMES = {
+    'IPv4': 'ipv4',
+    'IPv6': 'ipv6',
+    'DomainName': 'domain_name',
+    'File': 'file',
+    'Link': 'link',
+    'AndroidAab': 'android_aab',
+    'AndroidApk': 'android_apk',
+    'IOSIpa': 'ios_ipa',
+    'AndroidStore': 'android_store',
+    'IOSStore': 'ios_store',
+}
+
+
+class MetaDataType(enum.Enum):
+    """Asset Location MetaData Type"""
+    FILE_PATH = enum.auto()
+    CODE_LOCATION = enum.auto()
+    URL = enum.auto()
+    PORT = enum.auto()
+
+
+@dataclasses.dataclass
+class AssetLocationMetaData:
+    """Asset location metadata holding the type and the value of 'where' the vulnerability was detected in the asset """
+    type: MetaDataType
+    value: str
+
+
+@dataclasses.dataclass
+class AssetLocation:
+    """Asset location used to attach a vulnerability to a specific asset"""
+    metadata: List[AssetLocationMetaData]
+    asset: asset.Asset
+
+    def to_dict(self) -> dict:
+        """Convert data class to a dict matching what is expected from protobuf."""
+        return {
+            MAP_ASSETS_NAMES[self.asset.__class__.__name__]: self.asset.__dict__,
```

Comment:
I would've put the classes as keys of MAP_ASSETS_NAMES, & use type(self.asset) here.